### ISSUE TITLE
vktrace: vkFlushMappedMemoryRanges does not handle VK_WHOLE_SIZE

### ISSUE THREAD

**Issue by joey-lunarg** [LunarG/VulkanTools#144]
Tried using the following code to flush invalidated memory for tracing:

```
VkMappedMemoryRange range = {};
range.sType = VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE;
range.pNext = nullptr;
range.memory = frame_data_mem_;
range.offset = 0;
range.size = VK_WHOLE_SIZE;

vk::FlushMappedMemoryRanges(dev_, 1, &range);

// dev_ is a VkDevice, frame_data_mem_ is a VkDeviceMemory
```

The trace layer does not handle VK_WHOLE_SIZE (the magic number "18446744073709551615") and tries to memcpy that number of bytes, which causes a crash in the function "vktrace_pageguard_memcpy".

As a workaround, I just had to replace VK_WHOLE_SIZE with the actual size of the allocated memory.

**Comment by davidlunarg** [comment]
Attached is a quote from an email from Ming Zheng, the original author of this code, describing the fix.

[EmailOnFix.txt](https://github.com/LunarG/VulkanTools/files/702313/EmailOnFix.txt)

**Comment by davidlunarg** [comment]
Fixed in commit 5d2055dcd9b06e61d5dda1e9a4b8d78d758ce9cf
